### PR TITLE
searcher: Consolidate FetchTar and FetchTarPaths

### PR DESCRIPTION
**Pull request by eseliger**
There was a todo comment that said we want to consolidate them but didn't yet to keep another diff smaller - so now we're doing that.

Test plan: Integration tests for search still pass.

### CODE REVIEW

**Comment by eseliger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64268?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#64268** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64268?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#64261** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64261?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64260** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64260?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64259** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64259?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64159** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64159?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64158** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64158?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64157** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64157?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64156** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64156?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64155** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64155?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64071** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64071?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64070** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64070?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64069** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64069?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64068** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64068?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64067** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64067?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64066** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64066?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64065** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64065?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64064** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64064?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64063** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64063?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64062** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64062?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64061** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64061?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64058** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64058?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64056** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64056?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64055** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64055?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64054** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64054?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63655** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63655?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63651** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63651?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63650** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63650?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63649** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63649?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63648** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63648?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64012** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64012?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63743** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63743?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @eseliger and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cmd/searcher/internal/search/search_regex_test.go [hunk 1]
```diff
@@ -468,8 +468,8 @@ var githubStore = &Store{
 	ObservationCtx: &observation.TestContext,
 }
 
-func fetchTarFromGithub(ctx context.Context, repo api.RepoName, commit api.CommitID) (io.ReadCloser, error) {
-	r, err := fetchTarFromGithubWithPaths(ctx, repo, commit, []string{})
+func fetchTarFromGithub(ctx context.Context, repo api.RepoName, commit api.CommitID, paths []string) (io.ReadCloser, error) {
+	r, err := fetchTarFromGithubWithPaths(ctx, repo, commit, paths)
```
Comment (keegancsmith) [inline cmd/searcher/internal/search/search_regex_test.go:472]: you will need to run the benchmarks in this package to ensure you don't regress this behaviour.

Comment (eseliger) [reply cmd/searcher/internal/search/search_regex_test.go:472]: Uhm :( 

```
➜  sourcegraph git:(main) go test -bench=. ./cmd/searcher/internal/search/...
--- FAIL: BenchmarkColumnHelper
    chunk_test.go:374: column is not offset even though data is ASCII
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10623f684]

goroutine 5275 [running]:
github.com/sourcegraph/sourcegraph/cmd/searcher/internal/search.(*Store).fetch(0x108d98f00, {0x1070a2400, 0x1400221a570}, {0x1064c1142, 0x14}, {0x106554df7, 0x28}, 0x14002102100, {0x0, 0x0, ...})
        /Users/erik/Code/sourcegraph/sourcegraph/cmd/searcher/internal/search/store.go:272 +0x464
github.com/sourcegraph/sourcegraph/cmd/searcher/internal/search.(*Store).PrepareZip.func2.1({0x1070a2400?, 0x1400221a570?})
        /Users/erik/Code/sourcegraph/sourcegraph/cmd/searcher/internal/search/store.go:189 +0x5c
github.com/sourcegraph/sourcegraph/internal/diskcache.(*store).Open.func1({0x1070a2400?, 0x1400221a570?}, {0x140028bc4e0, 0x60})
        /Users/erik/Code/sourcegraph/sourcegraph/internal/diskcache/cache.go:113 +0x34
github.com/sourcegraph/sourcegraph/internal/diskcache.doFetch({0x1070a2400, 0x1400221a570}, {0x140028bc300, 0x5b}, 0x1400224a1c0, {0x1070af9c0, 0x1400298c410})
        /Users/erik/Code/sourcegraph/sourcegraph/internal/diskcache/cache.go:257 +0x30c
github.com/sourcegraph/sourcegraph/internal/diskcache.(*store).OpenWithPath.func2()
        /Users/erik/Code/sourcegraph/sourcegraph/internal/diskcache/cache.go:185 +0x218
created by github.com/sourcegraph/sourcegraph/internal/diskcache.(*store).OpenWithPath in goroutine 5274
        /Users/erik/Code/sourcegraph/sourcegraph/internal/diskcache/cache.go:172 +0x798
exit status 2
FAIL    github.com/sourcegraph/sourcegraph/cmd/searcher/internal/search 3.441s
FAIL
```

Seems like it's borked on main and that `fetchTarFromGithubWithPaths` never properly respected paths..

Comment (keegancsmith) [reply cmd/searcher/internal/search/search_regex_test.go:472]: Cool, go ahead and merge I will take a look in https://linear.app/sourcegraph/issue/SPLF-183/benchmarks-in-searcher-broken

**Pull request merged**